mapwms.c

```diff
@@ -347,25 +347,21 @@ int msWMSApplyFilter(mapObj *map, int version, const char *filter,
   if (filter[0] == '(') {
     paszFilters = FLTSplitFilters(filter, &numfilters);
 
-    if ( paszFilters && numfilters > 0 && numlayers != numfilters ) {
-      msFreeCharArray(paszFilters, numfilters);
-      paszFilters = NULL;
-    }
```

Comment:
Why was this block removed? Removing it creates a possible leak in case of error.

Reply:
As discussed, we longer want to validate that the number of wms layer match the number of filters. This is so because when we're using a single group of layer, we want only one filter to be apply to all layers inside the group.

Reply:
Agreed. I had misinterpreted those lines on my first review.